.gitlab-ci.yml

```diff
@@ -150,6 +150,18 @@ New OSTree:
       - RUNNER:
           - openstack/rhel-8.5-x86_64
 
+# Update once RHEL-8.5 runners are used for integration testing!
+.INTEGRATION_RULES: &INTEGRATION_RULES
+  rules:
+    - if: '$CI_PIPELINE_SOURCE != "schedule" && $SCRIPT != "libvirt.sh" || ($CI_PIPELINE_SOURCE != "schedule" && $SCRIPT == "libvirt.sh" && $RUNNER != "aws/centos-stream-8-x86_64")'
+    - if: '$CI_PIPELINE_SOURCE != "schedule" && $SCRIPT == "libvirt.sh" && $RUNNER == "aws/centos-stream-8-x86_64"'
+      variables:
+        RUNNER: "openstack/centos-stream-8-x86_64"
+    - if: '$CI_PIPELINE_SOURCE == "schedule" && $SCRIPT == "libvirt.sh" && $RUNNER == "aws/rhel-8.4-x86_64"'
+      variables:
+        RUNNER: "openstack/rhel-8.4-x86_64"
+    - if: '$CI_PIPELINE_SOURCE == "schedule" && $RUNNER =~ /[\S]+rhel-[8-9]\.[\S]+/ && $SCRIPT != "libvirt.sh"'
```

Comment:
Why are these rules so complicated ? What are you trying to achieve here ?

Instead of using all of this complexity why not simply create another stage/matrix definition for the libvirt test jobs and define only the runners which are known to work?

Reply:
You're right that is a lot simpler I just didn't want to make the matrix longer or duplicate some runners there so I was looking for a different way to achieve what I wanted which is run the libvirt test on openstack only in case of centos-stream-8 and rhel-8.5